GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -384,45 +341,46 @@ - (void)hasEventsForTarget:(GDTCORTarget)target onComplete:(void (^)(BOOL hasEve
 
 - (void)checkForExpirations {
   dispatch_async(_storageQueue, ^{
-    NSMutableSet<NSString *> *pathsToDelete = [[NSMutableSet alloc] init];
     GDTCORLogDebug(@"%@", @"Checking for expired events and batches");
     NSTimeInterval now = [NSDate date].timeIntervalSince1970;
     NSFileManager *fileManager = [NSFileManager defaultManager];
-    NSString *eventDataPath = [GDTCORFlatFileStorage eventDataStoragePath];
-    NSDirectoryEnumerator *enumerator = [fileManager enumeratorAtPath:eventDataPath];
-    NSString *path;
-    while ((path = [enumerator nextObject])) {
-      NSString *fileName = [path lastPathComponent];
-      NSDictionary<NSString *, id> *eventComponents = [self eventComponentsFromFilename:fileName];
-      NSDate *expirationDate = eventComponents[kGDTCOREventComponentsExpirationKey];
-      if (expirationDate != nil && expirationDate.timeIntervalSince1970 < now) {
-        [pathsToDelete addObject:[eventDataPath stringByAppendingPathComponent:path]];
-      }
-    }
 
-    // TODO: Events from expired batches with not expired events must be moved back to queue to
-    // avoid data loss.
     // TODO: Storage may not have enough context to remove batches because a batch may be being
     // uploaded but the storage has not context of it.
+
+    // Find expired batches and remove them moving the events back to the main storage.
```

Comment:
Do you mean 

"Find expired batches and move their events back to the main storage."

?

Reply:
yes, exactly what I mean, thank you! Sorry, I corrected the comment a couple times and ended up with a weird version.